packages/core/strapi/lib/types/core/attributes/relation.d.ts

```diff
@@ -1,48 +1,62 @@
-import { SchemaUID } from '../../utils';
-import { Attribute, ConfigurableOption, PrivateOption } from './base';
-import { GetAttributesByType, GetAttributesValues } from './utils';
+import type { Attribute, Common, Utils } from '@strapi/strapi';
 
-export type BasicRelationsType = 'oneToOne' | 'oneToMany' | 'manyToOne' | 'manyToMany';
-export type PolymorphicRelationsType = 'morphToOne' | 'morphToMany' | 'morphOne' | 'morphMany';
+export type BasicRelationsType =
+  | 'oneToOne'
+  | 'oneToMany'
+  | 'manyToOne'
+  | 'manyToMany'
+  | 'morphOne'
+  | 'morphMany';
+export type PolymorphicRelationsType = 'morphToOne' | 'morphToMany';
 export type RelationsType = BasicRelationsType | PolymorphicRelationsType;
 
-export interface BasicRelationAttributeProperties<
-  S extends SchemaUID,
-  R extends RelationsType,
-  T extends SchemaUID
-> {
+export type BasicRelationProperties<
+  S extends Common.UID.Schema,
+  R extends BasicRelationsType,
+  T extends Common.UID.Schema
+> = {
   relation: R;
   target: T;
-  inversedBy?: RelationsKeysFromTo<T, S>;
-  mappedBy?: RelationsKeysFromTo<T, S>;
-}
+} & R extends `morph${string}`
```

Comment:
Can we make `morph${string}` a type in itself so we can potentially use it elsewhere and it can be made safer than just looking at the prefix? (like explicitly check morphOne and morphMany)?

Reply:
Where would you see this type util live though? In the utilities? In this very file? In the "common.d.ts" next to this file?

Also, I'll change the definition to use `Utils.Suffix` here

Reply:
Also, since this is a type very specific (checking morph relation but only those that are not the main side), I don't know if it's worth creating a dedicated type for this (at least not one that is exported elsewhere. What do you think?)

Reply:
Yeah, not exported anywhere else. Just in here so it's more clear.